### PR TITLE
Add .editorconfig for csharp directory

### PR DESCRIPTION
**Pull request by Jensaarai**
Nicer experience for e.g. VS Code

Split from https://github.com/protocolbuffers/protobuf/pull/9666

### CODE REVIEW

File: csharp/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+# New line preferences
+# See https://github.com/protocolbuffers/protobuf/issues/9526
+#end_of_line = lf
```
Comment (amanda-tarafa) [inline csharp/.editorconfig:16]: Shouldn't this be uncommented?

**Comment by jskeet** [comment]
Please add the file to `Makefile.am` (along with the other C# files in `csharp_EXTRA_DIST`) - that will fix the Kokoro errors.

File: csharp/.editorconfig (hunk 1 above)
Comment (jskeet) [reply csharp/.editorconfig:16]: Not yet - we may well do that as part of fixing #9526, but this PR just adds the editor without trying to address that.

**Comment by Jensaarai** [comment]
Done...for my own curiosity, what is that file for? I'm guessing it's for packaging up the e.g. [protobuf-csharp-3.20.0.zip](https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protobuf-csharp-3.20.0.zip) zip/tar files on the releases, for users who want to consume as source?

**Comment by jskeet** [comment]
@Jensaarai: That's my understanding, yes. It was already in place when I started contributing to the project, and I haven't gone into great depths to check it :)

**Review by jskeet — APPROVED** [review]
Merging this now - it'll help with #9526.

**Pull request merged**